src/modules/AutoinstConfig.rb

```diff
@@ -578,6 +618,78 @@ def MainHelp
     publish :function => :ShellEscape, :type => "string (string)"
     publish :function => :AutoinstConfig, :type => "void ()"
     publish :function => :MainHelp, :type => "string ()"
+
+    private
+
+    # Reads base product name from the profile
+    #
+    # FIXME: Currently it returns first found product name. It should be no
+    # problem since this section was unused in AY installation so far.
+    # However, it might be needed to add a special handling for multiple
+    # poducts in the future
+    #
+    # @param [Hash] AY profile
+    # @return [String] product name
+    def base_product_name(profile)
+      software = profile.fetch("software", {})
+      software.fetch("products", {})["product"]
```

Comment:
/software/products is an array, not a hash

Reply:
hm. Not always. See http://paste.opensuse.org/98406650

Reply:
But what is the XML for that? Yours, or from a built product? IMHO it is invalid according to the schema.

Reply:
OK, it is https://github.com/mchf/aytests-tests/blob/3ee2df2fa4c3d150aea27090badc7920ed120f0d/aytests/sle15_product_selection_1.xml , but that is invalid. The `config:type=list`  attribute is missing:
```console
$ jing /usr/share/YaST2/schema/autoyast/rng/profile.rng sle15_product_selection_1.xml
...
/tmp/sle15_product_selection_1.xml:23:15: error: element "products" missing required attribute "config:type"
```

Reply:
ok will fix it.